pkg/apis/duck/v1alpha1/annotations.go

```diff
@@ -161,3 +169,25 @@ func validateAnnotationNotExists(annotations map[string]string, annotation strin
 	}
 	return errs
 }
+
+// ValidateClusterNameAnnotation validates the cluster-name annotation.
+func ValidateClusterNameAnnotation(annotations map[string]string, errs *apis.FieldError, client metadataClient.Client) *apis.FieldError {
+	if client.OnGCE() {
```

Comment:
I expected the KSA name for WI would include the cluster name, so it can avoid potential security issue like multiple clusters are using the same KSA name and Namespace name. (as all KSAs that share a name, Namespace name, and Workload Identity pool resolve to the same member name, and therefore share access to GSAs.)

Reply:
Besides that, I don't find other bad things...

Reply:
Another option is to avoid deleting the binding if no cluster name is present. This eliminates the problem of other clusters suddenly losing their credentials. However, it opens a security hole: a service account in a different cluster with a matching namespace and name will get the permissions of the bound GSA, even if no identity in the cluster ever had permissions to create that binding.